### ISSUE TITLE
Expose some modified MagmaCoreDatabase methods on MagmaCoreService

### ISSUE THREAD

**Issue by twalmsley** [gchq/MagmaCore#67]
A project I'm working on needs two additional methods as shown in the code below, with some slight impact on `MagmaCoreDatabase` and its implementers.

1. The first method is useful for finding all items with a certain predicate, such as `HQDM.ENTITY_NAME`, `HQDM.MEMBER_OF_KIND`, or `RDFS.RDF_TYPE`
2. The second method is a more specific version on the first for finding objects of a specific `RDFS.RDF_TYPE` or members of a specific `kind` etc.
```java
    /**
     * Find objects by a predicate.
     *
     * @param <T>       HQDM entity type.
     * @param predicate the predicate {@link HqdmIri}
     * @return a List of {@link Thing} that were found.
     */
    public <T extends Thing> List<T> findByPredicateIriOnly(final IRI predicate) {
        final List<T> searchResult = (List<T>) database.findByPredicateIriOnly(predicate);
        return searchResult;
    }

    /**
     * Find objects by a predicate value.
     *
     * @param <T>       HQDM entity type.
     * @param predicate the predicate {@link IRI}
     * @param value     The value of the predicate.
     * @return a List of {@link Thing} that were found.
     */
    public <T extends Thing> List<T> findByPredicateIriAndValue(final IRI predicate, final Object value) {
        final List<T> searchResult = (List<T>) database.findByPredicateIriAndValue(predicate, value);
        return searchResult;
    }
```

**Comment by twalmsley** [comment]
Waiting on som PRs so I can rebase this change and create another PR.
